### PR TITLE
Add mtl instances for all datatypes

### PR DESCRIPTION
**Pull request by 1Jajen1**
Also moves eq and genK instances because those need to be in incubator-test. Follow up for #19 

This also fixes a bug uncovered by this regarding `OptionT`'s `apEval` implementation

### CODE REVIEW

**Comment by aballano** [comment]
@1Jajen1 If you don't mind, would you mind trying to uncomment the ConcurrentLaws tests? This way we might be able to close #85 in one shot

**Comment by 1Jajen1** [comment]
> @1Jajen1 If you don't mind, would you mind trying to uncomment the ConcurrentLaws tests? This way we might be able to close #85 in one shot

Did that locally. Currently running the tests, but I get some failures in all of them. For all instances "Concurrent Laws: cancellable should run CancelToken on cancel" is timing out. Also two monad applicative consistency tests with eitherT fail. I'll attempt to fix the EitherT tests first...

**Comment by 1Jajen1** [comment]
So I got all the inconsistencies with EitherT and OptionT fixed, but the concurrent laws still fail locally on that one law as mentioned above. I left them uncommented, for now to see if it works in ci. But even so I think this needs some more work

**Comment by 1Jajen1** [comment]
So this probably needs some work in fx before we can uncomment the concurrent laws. Because this pr does so much else I'll leave them as they were before and undo that change.

**Comment by 1Jajen1** [comment]
@aballano @pakoito I changed the OptionT and EitherT applicative problems and left the concurrent laws as they were before because they are still flaky (actually they have been failing every time for me, so may be something else)

File: arrow-incubator-test/src/main/kotlin/arrow/mtl/test/laws/MonadWriterLaws.kt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package arrow.mtl.test
+package arrow.mtl.test.laws
```
[the remark this answers is not in the document]
Comment (1Jajen1) [reply arrow-incubator-test/src/main/kotlin/arrow/mtl/test/laws/MonadWriterLaws.kt]: Nope, the other incubator parts should probably follow this as well. The test packages should expose more than just laws, which is why I think this distinction makes sense.

**Pull request merged**